src/crypto/crypto.ts

```diff
@@ -165,28 +212,15 @@ export class Crypto extends AsyncOptionalCreatable<CryptoOptions> {
     }
 
     const tag = tokens[1];
-    const iv = tokens[0].substring(0, BYTE_COUNT_FOR_IV * 2);
-    const secret = tokens[0].substring(BYTE_COUNT_FOR_IV * 2, tokens[0].length);
-
-    return this.key.value((buffer: Buffer) => {
-      const decipher = crypto.createDecipheriv(ALGO, buffer.toString('utf8'), iv);
+    const iv = tokens[0].substring(0, IV_BYTES[this.getCryptoVersion()] * 2);
+    const secret = tokens[0].substring(IV_BYTES[this.getCryptoVersion()] * 2, tokens[0].length);
 
-      let dec;
-      try {
-        decipher.setAuthTag(Buffer.from(tag, 'hex'));
-        dec = decipher.update(secret, 'hex', 'utf8');
-        dec += decipher.final('utf8');
-      } catch (err) {
-        const error = messages.createError('authDecryptError', [(err as Error).message], [], err as Error);
-        const useGenericUnixKeychain =
-          env.getBoolean('SF_USE_GENERIC_UNIX_KEYCHAIN') || env.getBoolean('USE_GENERIC_UNIX_KEYCHAIN');
-        if (os.platform() === 'darwin' && !useGenericUnixKeychain) {
-          error.actions = [messages.getMessage('macKeychainOutOfSync')];
-        }
-        throw error;
-      }
-      return dec;
-    });
+    // When everything is v2, we can remove the else
+    if (this.isV2Crypto()) {
+      return this.decryptV2({ tag, iv, secret, tokens });
```

Comment:
what's the need for passing iv/secret to the decrypt methods?

For example, decryptV2 could calculate them from the tokens.  If they both took the tokens, there wouldn't be all the conditionals here.